tensorflow/examples/android/src/org/tensorflow/demo/CameraActivity.java

```diff
@@ -64,18 +63,24 @@ public void onRequestPermissionsResult(int requestCode, String permissions[], in
   }
 
   private boolean hasPermission() {
-    if (ContextCompat.checkSelfPermission(this, PERMISSION_CAMERA) == PackageManager.PERMISSION_DENIED || ContextCompat.checkSelfPermission(this, PERMISSION_STORAGE) == PackageManager.PERMISSION_DENIED) {
-      return false;
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      if (checkSelfPermission(PERMISSION_CAMERA) == PackageManager.PERMISSION_DENIED || checkSelfPermission(PERMISSION_STORAGE) == PackageManager.PERMISSION_DENIED) {
```

Comment:
No need for the blank line, and you can also do:
`return checkSelfPermission(PERMISSION_CAMERA) != PackageManager.PERMISSION_DENIED && checkSelfPermission(PERMISSION_STORAGE) != PackageManager.PERMISSION_DENIED;`
to remove the branching here.